### DISCUSSION TITLE
Secure Resource Access Control through API for Authorized Users

### DISCUSSION THREAD

**Discussion by jeffreymurdock** [pocketbase/pocketbase#1932] [Q&A] [answered]
Hello, pocketbase Team.

Need help with setting up API rules for resource access control. Example scenario:

-     Organizations have multiple users.
-     Projects belong to an organization.
-     Tickets belong to a project.

I am having trouble configuring the API rules, particularly for the 'Tickets' resource. Currently, I have '@request.auth.id != "" && @collection.project.organization.users ~ @request.auth.id' but it doesn't seem to be working. Can anyone assist with the correct syntax for this rule?

**Comment by ganigeorgiev** [comment, marked as answer]
The `@collection.*` filters is for targeting other collections that are not directly related to the current one (aka. when you don't have a `relation` field).

So in your case if for example `tickets` has a single `relation` field `project` (pointing to `projects`) then you can use something like:

```js
@request.auth.id != "" &&
project.organization.users.id ?= @request.auth.id
```
_(note that the `~` will still work but the "at least one of equal" operator is better for this case since it compares the ids 1:1)_
